Makefile

```diff
@@ -54,19 +72,21 @@ else
       $(warning OpenMP not found, skipping OpenMP support)
     endif
   else
-    # Check for OpenMP support in GCC or Clang on Linux
-    ifeq ($(shell echo | $(CC) -fopenmp -x c -E - > /dev/null 2>&1; echo $$?), 0)
-      CFLAGS += -fopenmp -DOMP
-      LDLIBS += -lgomp
-      $(info OpenMP found, compiling with OpenMP support)
-    else
-      $(warning OpenMP not found, skipping OpenMP support)
+    ifneq ($(OS), Windows_NT)
+      # Check for OpenMP support in GCC or Clang on Linux
+      ifeq ($(shell echo | $(CC) -fopenmp -x c -E - > /dev/null 2>&1; echo $$?), 0)
+        CFLAGS += -fopenmp -DOMP
+        LDLIBS += -lgomp
+        $(info OpenMP found, compiling with OpenMP support)
+      else
+        $(warning OpenMP not found, skipping OpenMP support)
+      endif
     endif
   endif
 endif
 
-# PHONY means these targets will always be executed
-.PHONY: all train_gpt2 test_gpt2 train_gpt2cu test_gpt2cu train_gpt2fp32cu test_gpt2fp32cu
+# DEFAULT means these targets will always be executed
+.DEFAULT: all 
```

Comment:
I have though about it and I think you should reinstate .PHONY but correct the comment above it so that it isn't misleading. In this case, our first recipe definition is already "all" which we intend it to be, so there is no need for DEFAULT. The reason as to why .PHONY is used is not this, it is so that Make does not care if the file exist or not. If it exists, it will be rebuilt anyway. The CI actually relies on this. Otherwise, one would have to throw in "make clean" in order to trigger new builds after for example one toggles OpenMP on/off.

Reply:
Got it - I was just trying to avoid the rebuild if it's already built. But I can put it back - no worries.